CMakeLists.txt

```diff
@@ -0,0 +1,218 @@
+    -Wno-strict-aliasing
+    -Wold-style-definition
+    -Wshadow -Waggregate-return
+    -Wstrict-prototypes
+    -Wuninitialized
+    -Wvla
+  )
+
+  if(CMAKE_C_COMPILER_ID MATCHES "Clang")
+      list(APPEND c_flags -Watomic-implicit-seq-cst -Wshorten-64-to-32)
+  endif()
+
+  add_compile_options(
+    "$<$<COMPILE_LANGUAGE:C>:${c_flags}>"
+  )
+endif()
+
+
+
+find_package(re CONFIG REQUIRED HINTS ../re/cmake)
+
+list(APPEND RE_DEFINITIONS
+  -DVERSION="${PROJECT_VERSION_FULL}"
+  -DVER_MAJOR=${PROJECT_VERSION_MAJOR}
+  -DVER_MINOR=${PROJECT_VERSION_MINOR}
+  -DVER_PATCH=${PROJECT_VERSION_PATCH}
+)
+
+add_definitions(${RE_DEFINITIONS})
+
+include_directories(
+  include
+  src
+  /usr/local/include
+  ${RE_INCLUDE_DIRS}
+  ${REM_INCLUDE_DIRS}
+  ${OPENSSL_INCLUDE_DIR}
+)
+
+link_directories(/usr/local/lib)
+
+if(MOD_PATH)
+  add_definitions(-DMOD_PATH="${MOD_PATH}")
+elseif(CMAKE_INSTALL_FULL_LIBDIR)
+  add_definitions(-DMOD_PATH="${CMAKE_INSTALL_FULL_LIBDIR}/baresip/modules")
+endif()
+
+if(SHARE_PATH)
+  add_definitions(-DSHARE_PATH="${SHARE_PATH}")
+else()
+  add_definitions(-DSHARE_PATH="${CMAKE_INSTALL_FULL_DATADIR}/baresip")
+endif()
+
+if(DEFAULT_CAFILE)
+  add_definitions(-DDEFAULT_CAFILE="${DEFAULT_CAFILE}")
+endif()
+
+if(DEFAULT_AUDIO_DEVICE)
+  add_definitions(-DDEFAULT_AUDIO_DEVICE="${DEFAULT_AUDIO_DEVICE}")
+endif()
+
+if(STATIC)
+  add_definitions(-DSTATIC)
+endif()
+
+
+##############################################################################
+#
+# Modules
+#
+
+include(modules)
+
+foreach(mod IN LISTS MODULES)
+  add_subdirectory(modules/${mod})
+  set_target_properties(${mod} PROPERTIES PREFIX "")
+endforeach()
+
+foreach(mod IN LISTS APP_MODULES)
+  add_subdirectory(
+    ${APP_MODULES_DIR}/${mod}
+    ${CMAKE_CURRENT_BINARY_DIR}/app_modules/${mod}
+  )
+  set_target_properties(${mod} PROPERTIES PREFIX "")
+  list(APPEND MODULES ${mod})
+endforeach()
+
+if(STATIC)
+  foreach(mod IN LISTS MODULES)
+    set(MOD_EXPORTS
+      "${MOD_EXPORTS}extern const struct mod_export exports_${mod};\n")
+    set(MOD_EXPORT_TABLE
+      "${MOD_EXPORT_TABLE}  &exports_${mod},\n")
+  endforeach()
+
+  configure_file(src/static.c.in src/static.c)
+  list(APPEND SRCS ${CMAKE_CURRENT_BINARY_DIR}/src/static.c)
+else()
+  foreach(mod IN LISTS MODULES)
+    if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+      set_target_properties(${mod} PROPERTIES
+                            LINK_FLAGS "-undefined dynamic_lookup")
+    endif()
+  endforeach()
+endif()
+
+
+##############################################################################
+#
+# Main target library
+#
+
+set(LINKLIBS ${REM_LIBRARIES} ${RE_LIBRARIES} Threads::Threads)
+
+if(ZLIB_FOUND)
+  list(APPEND LINKLIBS ZLIB::ZLIB)
+endif()
+
+if(USE_OPENSSL)
+  list(APPEND LINKLIBS OpenSSL::SSL OpenSSL::Crypto)
+endif()
+
+if(WIN32)
+  list(APPEND LINKLIBS ws2_32 iphlpapi winmm gdi32 crypt32 strmiids
+      ole32 oleaut32 qwave)
+else()
+  list(APPEND LINKLIBS -lm)
+endif()
```

Comment:
I think this section can be removed too. Linking should not necessary for modules.